### PR TITLE
Remove support for IE 11

### PR DESCRIPTION
**Pull request by carterworks**
> [!IMPORTANT]
> This PR must not be merged before gaining approval of all the stakeholders. They need to be aware that they will not be collecting data from IE and need to be okay with that. The question: "How much money does Adobe lose (or how many hits are left on the table) by dropping support for IE11?" needs to be answered.

## Description

- Remove IE 11 and 10 from the Babel transpiration targets
- Remove and re-write the utilities that worked around IE limitations

## Related Issue

[PDCL-11084](https://jira.corp.adobe.com/browse/PDCL-11084)

## Motivation and Context

Removing support for IE will decrease the library size and allow us to use more convenient syntax and features, like `Array.prototype.includes`, `async`/`await`, and array and object destructuring.

```
$ npm run clean && npm run build && gzip --keep dist/alloy.min.js && du -ah dist/

With IE10 and 11 target
 32K    dist/alloy.min.js.gz
 92K    dist/alloy.min.js
444K    dist/alloy.js
568K    dist

Without IE10 and 11 target
 28K    dist/alloy.min.js.gz
 84K    dist/alloy.min.js
428K    dist/alloy.js
540K    dist

…and without `utils/assign.js`, `utils/find.js`, `utils/includes.js` `utils/values.js`, rewriting `utils/querystring.js`, `utils/isNumber.js`, `utils/isUnique.js`, `utils/padStart.js`, and removing `@adobe/reactor-query-string`, `@adobe/reactor-object-assign`, and `css.escape`
 28K    dist/alloy.min.js.gz
 80K    dist/alloy.min.js
420K    dist/alloy.js
528K    dist
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -106,8 +106,8 @@ const buildConfig = (variant, minify) => {
           file: `${destDirectory}alloy${minifiedExtension}.js`,
           format: "iife",
           intro:
-            "if (document.documentMode && document.documentMode < 11) {\n" +
-            "  console.warn('The Adobe Experience Cloud Web SDK does not support IE 10 and below.');\n" +
+            "if (document.documentMode && document.documentMode <= 11) {\n" +
```
Comment (jonsnyder) [inline rollup.config.js:109]: This can just be `if (document.documentMode) {`

File: src/utils/isNumber.js [hunk 2]
```diff
@@ -16,4 +16,4 @@ governing permissions and limitations under the License.
  * @returns {boolean}
  */
 // eslint-disable-next-line no-restricted-globals
-export default value => typeof value === "number" && !isNaN(value);
+export default value => typeof value === "number" && !Number.isNaN(value);
```
Comment (jonsnyder) [inline src/utils/isNumber.js:19]: You can remove the eslint disable comment now that you've fixed this.

File: src/utils/querystring.js [hunk 3]
```diff
@@ -9,7 +9,17 @@ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTA
...
+    let parsedString = string;
+    if (typeof string === "string") {
+      parsedString = string.trim().replace(/^[?#&]/, "");
+      parsedString = decodeURIComponent(parsedString);
+    }
+    return Object.fromEntries(new URLSearchParams(parsedString).entries());
+  },
+  stringify: object =>
+    new URLSearchParams(object).toString().replace("+", "%20")
+};
```
Comment (jonsnyder) [inline src/utils/querystring.js:25]: Why not use URL constructor to pull out the query part?

**Review by jonsnyder — APPROVED** [review]

File: src/utils/querystring.js (hunk 3 above)
Comment (carterworks) [reply src/utils/querystring.js:25]: `URL` only parses complete urls with the subdomain, domain, and TLD. So `new URL(location.search)`, like we do in `injectAddQueryStringIdentityToPayload.js` would throw an error. This was is more flexible.

**Comment by carterworks** [comment]
There are over 900 merge conflicts.

I am going to close this PR and recreate it based off of the most recent commit.

**Pull request closed without merging**